### PR TITLE
Making the list terminator incorporate the label instead of just the separator.

### PR DESCRIPTION
**Pull request by radumereuta**
@bmmoore please review.

### CODE REVIEW

File: src/javasources/KTool/src/org/kframework/compile/transformers/FlattenSyntax.java [hunk 1]
```diff
@@ -32,13 +31,13 @@ public ASTNode visit(Definition node, Void _)  {
...
+        listUnits.clear();
         node = (Module) super.visit(node, _);
-        if (listSeparators.isEmpty())
+        if (listUnits.isEmpty())
             return node;
 
-        for (String sep : listSeparators) {
-            node.addConstant(Sort.KLABEL, MetaK.getListUnitLabel(sep));
+        for (String listUnit : listUnits) {
+            node.addConstant(Sort.KLABEL, listUnit);
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/compile/transformers/FlattenSyntax.java:40]: what is happening here?

File: src/javasources/KTool/src/org/kframework/compile/transformers/FlattenTerms.java [hunk 2]
```diff
@@ -96,7 +96,7 @@ public ASTNode visit(TermCons tc, Void _)  {
             }
             String label;
             if (tc.isListTerminator())
-                label = tc.getProduction().getListDecl().getTerminatorKLabel();
+                label = tc.getProduction().getTerminatorKLabel();
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/compile/transformers/FlattenTerms.java:99]: this doesn't feel like a good api to me. Stuff specific to user lists should be on the UserList class.

File: src/javasources/KTool/src/org/kframework/compile/transformers/FlattenSyntax.java (hunk 1 above)
Comment (radumereuta) [inline src/javasources/KTool/src/org/kframework/compile/transformers/FlattenSyntax.java:40]: I just made a mechanical transformation, but I assume is needed to declare all the list terminators as KLabels to make maude parse it. Traian wrote this code, so if you are interested in knowing more, you should ask him.

File: src/javasources/KTool/src/org/kframework/compile/transformers/FlattenTerms.java (hunk 2 above)
Comment (radumereuta) [inline src/javasources/KTool/src/org/kframework/compile/transformers/FlattenTerms.java:99]: That's the problem I need information from the Production (namely the klabel). I just need to know that I have a UserList inside, but I'm not interested in any of it's information.
This is the right place to put it.

**Pull request merged**